calico_kubernetes/calico_kubernetes.py

```diff
@@ -147,6 +165,117 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _container_add(self, container_name, workload_id, pid, ip, interface,
+                       hostname, orchestrator_id):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+
+        :param container_name: The name or ID of the container.
+        :param ip: An IPAddress object with the desired IP to assign.
+        :param interface: The name of the interface in the container.
+        """
+        # Check if the container already exists. If it does, exit.
+        try:
```

Comment:
A cleaner, more readable way to do this is to use get_endpoints, which will return an empty list instead of raising an error.
```
if len(self._datastore_client.get_endpoint(
                hostname=hostname,
                orchestrator_id=orchestrator_id,
                workload_id=workload_id)) == 1:
      print("This container has already been configured with Calico Networking.")
      sys.exit(1)
```